app/api/controllers/RequestHandler.scala

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package api.controllers
+
+import api.hateoas.{HateoasData, HateoasFactory, HateoasLinksFactory, HateoasWrapper}
```

Comment:
Probably not important enough to go back through everything, but I don't think you have scalafmt configured for this repo, it would usually format imports with spaces between the braces.

Reply:
scalafmt's definitely configured for it (made me check though, phew!). I've noticed with some of these APIs, formatting a whole folder from the Project explorer does the imports like the above. Probably a slight difference in the scalafmt files somewhere, or perhaps the version.

Reply:
I think what's happening is that Intellij is overriding the scalafmt rules. The rule in scalafmt is

```
spaces.inImportCurlyBraces: true
```

so the spaces should definitely be there. There's this [issue](https://youtrack.jetbrains.com/issue/SCL-20791) I found which describes the problem, essentially Intellij's optimise import functionality doesn't respect the scalafmt rules, so if you click reformat code on a folder and run it with the 'Optimise imports' rule selected, it first runs the scalafmt rules and then changes the imports again based on its built in rules (which I don't think can be changed).